TodoLogic/src/components/Todo.tsx

```diff
@@ -1,75 +1,128 @@
 import TodoList from "./TodoList";
 import TodoDate from "./TodoDate";
 import "./App.css";
-import { useState, useEffect, useCallback } from "react";
+import { useState, useEffect, createContext } from "react";
+interface TodoListProp {
+  id: number;
+  data: string;
+  check: boolean;
+}
+type contexttype = {
+  submit: TodoListProp[];
+  setSubmit: React.Dispatch<React.SetStateAction<TodoListProp[]>>;
+};
+//creating Context
+export const TodoContext = createContext({} as contexttype);
+
+const getLocalItems = () => {
+  let storageList = localStorage.getItem("submit");
 
-const Todo = (): JSX.Element => {
+  if (storageList) {
+    return JSON.parse(localStorage.getItem("submit"));
```

Comment:
Why localStorage.getItem is queried with same argument where it has already been stored inside a variable above